File: Example/ObjC/Bulletin/FeedbackPageBulletinItem.m
/**
 *  BulletinBoard
 *  Copyright (c) 2017 - present Alexis Aubry. Licensed under the MIT license.
 */

#import "FeedbackPageBulletinItem.h"

@interface FeedbackPageBulletinItem ()

@property (nonatomic, strong) UISelectionFeedbackGenerator *feedbackGenerator;

@end

@implementation FeedbackPageBulletinItem

- (void)actionButtonTappedWithSender:(UIButton *)sender
{
    // Play an haptic feedback
    [self.feedbackGenerator prepare];
    [self.feedbackGenerator selectionChanged];

    // Call super
    [super actionButtonTappedWithSender:sender];
}

- (void)alternateActionButtonTappedWithSender:(UIButton *)sender
{
    // Play an haptic feedback
    [self.feedbackGenerator prepare];
    [self.feedbackGenerator selectionChanged];

    // Call super
    [super alternateActionButtonTappedWithSender:sender];
}

@end

File: Example/Swift/Bulletin/FeedbackPageBulletinItem.swift
/**
 *  BulletinBoard
 *  Copyright (c) 2017 Alexis Aubry. Licensed under the MIT license.
 */

import UIKit
import BulletinBoard

/**
 * A subclass of page bulletin item that plays an haptic feedback when the buttons are pressed.
 *
 * This class demonstrates how to override `PageBLTNItem` to customize button tap handling.
 */

class FeedbackPageBLTNItem: PageBLTNItem {

    private let feedbackGenerator = SelectionFeedbackGenerator()

    override func actionButtonTapped(sender: UIButton) {

        // Play an haptic feedback

        feedbackGenerator.prepare()
        feedbackGenerator.selectionChanged()

        // Call super

        super.actionButtonTapped(sender: sender)

    }

    override func alternativeButtonTapped(sender: UIButton) {

        // Play an haptic feedback

        feedbackGenerator.prepare()
        feedbackGenerator.selectionChanged()

        // Call super

        super.alternativeButtonTapped(sender: sender)

    }

}
